### ISSUE TITLE
[profiles] Add existing profile "Gain-Offset" from ModBus binding to core to allow value scaling

### ISSUE THREAD

**Issue by Oggerschummer** [openhab/openhab-core#3538]
When it is necessary to adjust incoming or outgoing values it is currently possible to define an offset on the original value using profile "Offset".

Unfortunately some devices deliver float values scaled to integers, e.g. ModBus energy meters do so.
For that reason there is a [profile contained in the ModBus binding called "Gain-Offset"](https://next.openhab.org/addons/bindings/modbus/#profiles) that allows not only an offset but to provide a gain factor plus adding a UoM as well and works on incoming and outbound numeric values.

Clear advantage: No need to involve scripts to scale the original value, complete setup is possible from the UI.

See [this community post](https://community.openhab.org/t/item-number-change-decimal-point/145873/7) for more details on how this would simplify to end users life.

Either the existing profile "Offset" should be enhanced with a gain factor or it could even be superseded by the "Gain-Offset" profile.

OH Version 3.4, similar in OH 4 M1.
OS version independent.

**Comment by jimtng** [comment]
I'm working on adding a multiplier option to the existing system offset profile. I think there will be some differences in behaviour though.

To me it makes more sense to implement it as y = MX + b. 

I'm still pondering about the application of unit in the gain parameter.

**Comment by jimtng** [comment]
Value=1, gain= 0.3, offset=???

What is the output you're expecting?

**Comment by Oggerschummer** [comment]
Sorry, some clarification:
"???”: Represents the offset required to set the result to “0”
But as 1 * 0.3 results in 3.33333333… no value for the offset can be set output exactly "0" after scaling.
E.g. an offset of 3.3 still leaves 0.033333… as the result.

If we apply the offset first and then gain factor it is no problem at all:
Offset (aka "???”) = 1, gain= 0.3 -> (1-1)*0.3 = 0

**Comment by jimtng** [comment]
The modbus gain-offset profile uses the formula of (value + offset) * gain, or y = m(x + b) where m is the gain, and b is the offset.

[The more common linear equation](https://www.google.com/search?q=linear+equation) is `y = mx + b`. You can migrate (if you chose to do so - because I don't think we should remove the profile from modbus?) by simply multiplying your (modbus offset x the gain factor) to get the offset value to use in the new profile.

Example:
modbus profile (gain: 0.1, offset: 5) -> core profile (multiplier: 0.1, offset: 0.5)
modbus profile (gain: 10, offset: 0.5) -> core profile (multiplier: 10, offset: 5)

**Comment by jimtng** [comment]
> But as 1 * 0.3 results in 3.33333333

1 * 0.3 = 0.3, not 0.333333

<img width="665" alt="image" src="https://user-images.githubusercontent.com/2554958/231419171-d22a7714-6d8d-4a13-83be-00c94f8dee36.png">

**Comment by Oggerschummer** [comment]
Sorry, as said before - my calculation was wrong due to being a but overworked (And getting older). You are totally right, more common ist y= mx+b. 
Still the questions then is if we need this twice in (one profile in the binding, the other in the core) where the core version should be enough when finished.
But for now extending the existing core profile is definitely an improvement users will be thankful for.

**Comment by jimtng** [comment]
It's good to question and provide feedback. I appreciate it. There are two directions of calculation, one to find y (from handler to item) and the inverse, to find x (from item to handler). I think you might have mixed them up.

This is what the modbus gain-offset does for item to handler (the inverse of `y = m(x + b)` which is `x = y/m - b` so y is the input here, x is the output): input = 1, gain = 0.3, offset = 1 -> the output would be `1 / 0.3 - 1` = 3.3333333 - 1 = `2.333333`

Although it's evident from modbus profile's source code, to be absolutely sure, I wrote an actual custom test against the modbus profile:

```java
    @Test
    public void customInputOutputTest() {
        ProfileCallback callback = mock(ProfileCallback.class);
        ModbusGainOffsetProfile<?> profile = createProfile(callback, "2", "1");

        DecimalType input = new DecimalType(1);
        profile.onCommandFromHandler(input);
        ArgumentCaptor<Command> capture = ArgumentCaptor.forClass(Command.class);
        verify(callback, times(1)).sendCommand(capture.capture());
        logger.warn("XXXX onCommandFromHandler: {} -> {}", input, capture.getValue());

        profile.onCommandFromItem(input);
        capture = ArgumentCaptor.forClass(Command.class);
        verify(callback, times(1)).handleCommand(capture.capture());
        logger.warn("XXXX onCommandFromItem: {} -> {}", input, capture.getValue());
    }
```

So, input = 1, gain = 2, offset = 1.
Here's the result, which confirmed what I said above
```
main] WARN  o.o.b.m.i.p.ModbusGainOffsetProfileTest - XXXX onCommandFromHandler: 1 -> 4
[main] WARN  o.o.b.m.i.p.ModbusGainOffsetProfileTest - XXXX onCommandFromItem: 1 -> -0.5
```

In other words, the modbus profile formula is indeed: 
- `y = m(x+b)` for handler to item and 
- `x = y/m - b` for item to handler

The numbers:
- Handler to item y = 2 * (1 + 1) = `4` - which matched the log output above
- Item to handler: x = 1 / 2 - 1 = `- 0.5` = also matched the log output above

So there's the proof.

To convert into y = mx + b style of offset, as I said above:
 modbus: (gain = 2, offset = 1) -> core (gain = 2, offset = 2)

To get that same result  using core's profile formula, multiply the offset by the gain, so (gain = 2, offset = 2)
- Handler to item: y = 2 * 1 + 2 = 4
- Item to handler: x = (y - b) / m = (1 - 2) / 2 = - 0.5 

Same result

**Comment by jimtng** [comment]
@Oggerschummer do you think this is still needed?

**Comment by Oggerschummer** [comment]
> @Oggerschummer do you think this is still needed?

Guess not, closing.
